homeassistant/components/wiz/light.py

```diff
@@ -0,0 +1,358 @@
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.debug("Kelvin is not present in the library. Fallback to 6500")
+            return color_utils.color_temperature_kelvin_to_mired(6500)
+
+    def get_max_mireds(self) -> int:
+        """Return the warmest color_temp that this light supports."""
+        if self._bulbtype is None:
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+        # DW bulbs have no kelvin
+        if self._bulbtype.bulb_type == BulbClass.DW:
+            return 0
+        # If bulbtype is TW or RGB then return the kelvin value
+        try:
+            return color_utils.color_temperature_kelvin_to_mired(
+                self._bulbtype.kelvin_range.min
+            )
+        except WizLightNotKnownBulb:
+            _LOGGER.debug("Kelvin is not present in the library. Fallback to 2200")
+            return color_utils.color_temperature_kelvin_to_mired(2200)
+
+    def get_supported_features(self) -> int:
+        """Flag supported features."""
+        if not self._bulbtype:
+            # fallback
+            return SUPPORT_FEATURES_RGB
+        features = 0
+        try:
+            # Map features for better reading
+            if self._bulbtype.features.brightness:
+                features = features | SUPPORT_BRIGHTNESS
+            if self._bulbtype.features.color:
+                features = features | SUPPORT_COLOR
+            if self._bulbtype.features.effect:
+                features = features | SUPPORT_EFFECT
+            if self._bulbtype.features.color_tmp:
+                features = features | SUPPORT_COLOR_TEMP
+            return features
+        except WizLightNotKnownBulb:
+            _LOGGER.warning(
+                "Bulb is not present in the library. Fallback to full feature"
+            )
+            return SUPPORT_FEATURES_RGB
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects.
+
+        URL: https://docs.pro.wizconnected.com/#light-modes
+        """
+        return self._scenes
+
+    @property
+    def available(self):
+        """Return if light is available."""
+        return self._available
+
+    async def async_update(self):
+        """Fetch new state data for this light."""
+        await self.update_state()
+
+        if self._state is not None and self._state is not False:
+            self.update_brightness()
+            self.update_temperature()
+            self.update_color()
+            self.update_effect()
+
+    @property
+    def device_info(self):
+        """Get device specific attributes."""
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self._mac)},
+            "name": self._attr_name,
+            "manufacturer": "WiZ Light Platform",
+            "model": f"{self._bulbtype.name} Mac: {self._mac}",
```

Comment:
Mac is not part of the model, for debug purposes we should use diagnostics

```suggestion
            "model": self._bulbtype.name",
```